telegram/request/_requestdata.py

```diff
@@ -51,29 +56,29 @@ def __init__(self, parameters: Optional[List[RequestParameter]] = None):
 
     @property
     def parameters(self) -> Dict[str, Union[str, int, List[Any], Dict[Any, Any]]]:
-        """Gives the parameters as mapping of parameter name to the parameter value, which can be
-        a single object of type :obj:`int`, :obj:`float`, :obj:`str` or :obj:`bool` or any
-        (possibly nested) composition of lists, tuples and dictionaries, where each entry, key
-        and value is of one of the mentioned types.
+        """
+        Gives the parameters as mapping of parameter name to the parameter value.
 
         Returns:
             Dict[:obj:`str`, Union[:obj:`str`, :obj:`int`, List[any], Dict[any, any]]]
         """
+        
         return {
-            param.name: param.value  # type: ignore[misc]
+            param.name: param.value
             for param in self._parameters
-            if param.value is not None
+            if param.value is not None and isinstance(param.value, (str, int, list, dict))
+        } or {
+            param.name: logging.warning(
+                "Unexpected type for the value of '%s': %s", param.name, type(param.value)
+            ) or param.value
+            for param in self._parameters
+            if param.value is not None and not isinstance(param.value, (str, int, list, dict))
```

Comment:
this doesn't make any sense.

* You are doing somethin like `dict_1 or dict_2` so that the `isinstance` check within dict_1/dict_2 don't have much effect at all
* `param.name: logging.warning(…)` means that the value of that dict entry is the return value of `logging.warning(…)` which holds to meaning to the user at all
* must importantly you have shown no evidance that path 1) in https://github.com/python-telegram-bot/python-telegram-bot/pull/4508/files#r1784614370 is actually the way to go rather than path 2)